frontend/tests/components/auth/Login.test.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import { render, screen, waitFor } from '@testing-library/react';
+import { BrowserRouter } from 'react-router-dom';
+import axios from 'axios';
+import userEvent from '@testing-library/user-event'
+import Login from '../../../src/components/auth/Login';
+import { AUTH_BASE_URI } from '../../../src/constants/api';
+
+
+jest.mock('axios');
+
+describe('Login component', () => {
+
+  // mock login request
+  axios.post.mockImplementation({ data: { 
+      expiry: "2023-02-09T14:52:04.877168Z",
+      token: "12345a6680364c7bf58b54b8c3d59b93",
+      user: {email: "test@test.com", first_name: "test", full_name: "test user", last_name: "user", username: "test_user"},
+    }}
+  );
+
+  test('User login', async () => {
+    // mock user interaction: reccomanded to put this at the start of the test
+    const user = userEvent.setup();
+
+    render (
+      <BrowserRouter>
+        <Login />
+      </BrowserRouter>
+    )
+
+    // page before login
+    const usernameInputElement = screen.getByLabelText('Username');
+    expect(usernameInputElement).toBeInTheDocument();
```

Comment:
Unuseful `toBeInTheDocument` statements, since `getBy` throw exception if no element is found (see [here](https://testing-library.com/docs/queries/about#types-of-queries) ).
If you wish, you can mantain if for clarity

Reply:
Thanks for the suggestion! I read a tutorial and I discovered this behavior of `getBy` function, I thought the `explicit` assertion with the `toBeInTheDocument` is clearer.